### PR TITLE
deals with #104, check oauth scope record exists before writing to collection

### PR DESCRIPTION
**Pull request by RoboSparrow**
Console command looks now like this:

```bash
# first call
./X setup:oauth
Setting up default OAuth scopes...
  - new scope super added.
  - new scope statements/write added.
  - new scope statements/read added.
  - new scope statements/read/mine added.
  - new scope attachments added.
  - new scope state added.
  - new scope profile added.
  - new scope define added.
  - new scope all/read added.
  - new scope all added.
OAuth scopes configured!

# second call
./X setup:oauth
Setting up default OAuth scopes...
  - skip scope super exits already.
  - skip scope statements/write exits already.
  - skip scope statements/read exits already.
  - skip scope statements/read/mine exits already.
  - skip scope attachments exits already.
  - skip scope state exits already.
  - skip scope profile exits already.
  - skip scope define exits already.
  - skip scope all/read exits already.
  - skip scope all exits already.
OAuth scopes configured!

# removed "super" from collection and called again

./X setup:oauth
Setting up default OAuth scopes...
  - new scope super added.
  - skip scope statements/write exits already.
  - skip scope statements/read exits already.
  - skip scope statements/read/mine exits already.
  - skip scope attachments exits already.
  - skip scope state exits already.
  - skip scope profile exits already.
  - skip scope define exits already.
  - skip scope all/read exits already.
  - skip scope all exits already.
OAuth scopes configured!
```

### CODE REVIEW

**Comment by RoboSparrow** [comment]
@sraka1 , can you please action this. I have another PR in queue in regards to the `since` validation

File: src/xAPI/Console/BasicTokenCreateCommand.php [hunk 1]
```diff
@@ -79,6 +79,15 @@ protected function execute(InputInterface $input, OutputInterface $output)
             '',
         ]);
 
+        // get permissions from user service. Abort if no permissions set
+        $userService->fetchAvailablePermissions();
+        $hasPermissions = $userService->getCursor()->count();
+        if(!$hasPermissions){
```
Comment (sraka1) [inline src/xAPI/Console/BasicTokenCreateCommand.php]: PSR-2!

File: src/xAPI/Console/SetupDbCommand.php [hunk 2]
```diff
@@ -58,26 +61,44 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+        $output->writeln('<info>x</info> Configuration saved!');
+
+        // oAuth scopes
+        // @TODO retireve collection name from service
+        $output->writeln('Setting up default OAuth scopes...');
+        $mongo = new \API\Util\MongoClient($newConfig);
+        $collection = $mongo->db->getCollection('authScopes');
+        foreach($newConfig['xAPI']['supported_auth_scopes'] as $scope){
+            $exists = $collection->find()->where('name', $scope['name'])->findOne();
+            if($exists){
```
Comment (sraka1) [inline src/xAPI/Console/SetupDbCommand.php]: PSR-2!

File: src/xAPI/Console/SetupOAuthCommand.php [hunk 3]
```diff
@@ -44,9 +44,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $output->writeln('<info>Setting up default OAuth scopes...</info>');
 
         $oAuthService = new OAuthService($this->getSlim());
-
         foreach ($this->getSlim()->config('xAPI')['supported_auth_scopes'] as $authScope) {
             $scope = $oAuthService->addScope($authScope['name'], $authScope['description']);
+            if(!$scope){
```
Comment (sraka1) [inline src/xAPI/Console/SetupOAuthCommand.php]: PSR-2!

File: src/xAPI/Service/Auth/OAuth.php [hunk 4]
```diff
@@ -207,13 +207,16 @@ public function addScope($name, $description)
     {
         $collection  = $this->getDocumentManager()->getCollection('authScopes');
 
+        // #104, check if scope document record exists already
+        $exists = $collection->find()->where('name', $name)->findOne();
+        if($exists){
```
Comment (sraka1) [inline src/xAPI/Service/Auth/OAuth.php]: PSR-2!

**Comment by sraka1** [comment]
PR looks good, just needs some style fixes. You could probably run 'php-cs-fixer' on the files and it'll fix those up for you...

**Comment by RoboSparrow** [comment]
will do

**Pull request merged**